src/analysis/inner_product_space/pi_L2.lean

```diff
@@ -299,6 +333,68 @@ protected lemma coe_mk (hon : orthonormal 𝕜 v) (hsp: submodule.span 𝕜 (set
   ⇑(orthonormal_basis.mk hon hsp) = v :=
 by classical; rw [orthonormal_basis.mk, _root_.basis.coe_to_orthonormal_basis, basis.coe_mk]
 
+open submodule
+
+/-- A finite orthonormal family of vectors whose span has trivial orthogonal complement is an
+orthonormal basis. -/
+protected def mk_of_orthogonal_eq_bot (hon : orthonormal 𝕜 v) (hsp : (span 𝕜 (set.range v))ᗮ = ⊥) :
+  orthonormal_basis ι 𝕜 E :=
+orthonormal_basis.mk hon
+begin
+  haveI : finite_dimensional 𝕜 (span 𝕜 (range v)) :=
+    finite_dimensional.span_of_finite 𝕜 (finite_range v),
+  haveI : complete_space (span 𝕜 (range v)) := finite_dimensional.complete 𝕜 _,
+  rwa orthogonal_eq_bot_iff at hsp,
+end
+
+@[simp] protected lemma coe_of_orthogonal_eq_bot_mk (hon : orthonormal 𝕜 v)
+  (hsp : (span 𝕜 (set.range v))ᗮ = ⊥) :
+  ⇑(orthonormal_basis.mk_of_orthogonal_eq_bot hon hsp) = v :=
+orthonormal_basis.coe_mk hon _
+
+variable {ι' : Type*}
+variables [fintype ι']
+
+/-- If the underlying vector spaces are equivalent, the euclidean spaces built on top are too. -/
+def _root_.linear_equiv.to_linear_equiv_of_euclidean_space (e : (ι →₀ 𝕜) ≃ₗ[𝕜] (ι' →₀ 𝕜)) :
+  euclidean_space 𝕜 ι ≃ₗ[𝕜] euclidean_space 𝕜 ι' :=
+(finsupp.linear_equiv_fun_on_fintype 𝕜 𝕜 ι).symm.trans $
+  e.trans (finsupp.linear_equiv_fun_on_fintype 𝕜 𝕜 ι')
```

Comment:
Could you omit this for now?  I'd like to discourage people from finsupp in relation to Euclidean space, so let's not include it until we have a compelling use case.

Reply:
For sure. To be honest, I'm not sure why I have it in there at all.